homeassistant/components/screenlogic/number.py

```diff
@@ -64,7 +65,9 @@ async def async_set_value(self, value: float) -> None:
             levels[level] = self.coordinator.data[SL_DATA.KEY_SCG][level]["value"]
         levels[self._data_key] = int(value)
 
-        if await self.coordinator.gateway.async_set_scg_config(
+        if await cast(
+            ScreenlogicDataUpdateCoordinator, self.coordinator
+        ).gateway.async_set_scg_config(
```

Comment:
This is kinda ugly. Can we do 

`coordinator: ScreenlogicDataUpdateCoordinator` above instead?

Reply:
The `cast` is still needed, but I've moved it to the top of the `async_set_value`